manifests/2.4.0/opensearch-dashboards-2.4.0.yml

```diff
@@ -9,49 +9,49 @@ ci:
 components:
   - name: OpenSearch-Dashboards
     repository: https://github.com/opensearch-project/OpenSearch-Dashboards.git
-    ref: '2.4'
+    ref: '061e5bd8706c3a6a1aa089e6490c56397aa6b60a'
```

Comment:
Mistake?
Could not find this ref at all.
https://github.com/opensearch-project/opensearch-build/actions/runs/3399364426/jobs/5654890776

Reply:
Strange. Double checked the commit ID from https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/2.4.0/4275/linux/x64/tar/dist/opensearch-dashboards/manifest.yml and it's the same.

Reply:
The code is trying to checkout out using --branch with --depth 1 which does not allow commitID it seems. Only a branch or tag is allowed.

Let me see if we can change the code a bit to allow a full clone then git checkout.

Reply:
Interesting it works for the OS not the OSD one.
Need more debugging as this command is actually possible to checkout commits:
`git fetch --depth 1 origin 77ef9e304dd6ee95a600720a387a9735bbcf7bc9`

Reply:
OS: https://github.com/opensearch-project/opensearch-build/actions/runs/3399364426/jobs/5654890705
OSD: https://github.com/opensearch-project/opensearch-build/actions/runs/3399364426/jobs/5654890776

Seems like it is additional checks here that cause the issue due to `git ls-remote` will only check for branch/tag.
` /home/runner/work/opensearch-build/opensearch-build/src/ci_workflow/ci_check_list_source_ref.py`

Reply:
I fixed it in this PR @zelinh please review:
* https://github.com/opensearch-project/opensearch-build/pull/2856

Reply:
Just have above PR merged and fetch the changes. Re-running the CIs now.